plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -25,13 +35,50 @@ def extract_filepath(provided_regex: str, provided_string: str, provided_file: s
         new_string = regex.sub(Regex.Date, "", new_string)
         matches = regex.findall(provided_regex, new_string)
     elif provided_file:
-        new_file = base64.b64decode(provided_file.encode("utf-8")).decode("utf-8")
+        try:
+            new_file = base64.b64decode(provided_file.encode("utf-8")).decode("utf-8")
+        except UnicodeDecodeError:
+            new_file = extract_content_from_file(base64.b64decode(provided_file))
         new_file = regex.sub(Regex.URL, "", new_file)
         new_file = regex.sub(Regex.Date, "", new_file)
         matches = regex.findall(provided_regex, new_file)
     return list(dict.fromkeys(matches))
 
 
+def extract_content_from_file(provided_file: bytes) -> str:
+    with io.BytesIO(provided_file) as f:
+        try:
+            # extracting content from DOCX, PPTX, XLSX, ODT, ODP, ODF files
+            with zipfile.ZipFile(f) as unzip_files:
+                files_content = ""
+                x = unzip_files.infolist()
+                for i in enumerate(x):
+                    try:
+                        files_content += unzip_files.read(x[i[0]]).decode("utf-8")
+                    # After unpacking, may contain images for which the decoding will fail
+                    except UnicodeDecodeError:
+                        continue
+            # remove xml tags from files content
+            return regex.sub(r"<[\p{L}\p{N}\p{Lo}\p{So} :\/.\"=_%,(){}+#&;?-]*>", " ", files_content)
+        except zipfile.BadZipFile:
+            try:
+                # extracting content from PDF file
+                pdf_file = pdfplumber.open(f)
+                pages = pdf_file.pages
+                pdf_content = ""
+                for page in enumerate(pages):
+                    pdf_content += page[1].extract_text()
+                pdf_file.close()
+                return pdf_content
+            except PDFSyntaxError:
+                raise PluginException(
+                    cause="The type of the provided file is not supported.",
+                    assistance="Supported file types: PDF, DOCX, PPTX, XLSX, ODT, ODP, ODF",
+                )
+        finally:
+            f.close()
```

Comment:
I do not think that we need this `close()` call here. Opening/closing this `bytesIO` file stream should be handled automatically by Python when using the `with io.BytesIO(provided_file) as f:` line.

Reply:
You're right. I removed it.